src/node.rs

```diff
@@ -598,7 +622,12 @@ impl Node {
             0,
         ));
 
-        Ok(invoice)
+        if let Some(lsp) = self.lsp_client.clone() {
+            let invoice = lsp.get_lsp_invoice(invoice.to_string()).await?;
+            Ok(Invoice::from_str(&invoice)?)
```

Comment:
we might want to make sure that we are connected to the LSP, otherwise the LSP might be down.

Also if the `get_lsp_invoice` fails, we would just fail here, do you think that is okay? In some cases I can see that being  correct (if we have no channels) in others that would kinda suck (we have enough inbound)

Reply:
Fixed the first one by doing a connect_peer first, but it should at least be trying to auto connect when I added it to the node initialization.

> Also if the get_lsp_invoice fails, we would just fail here, do you think that is okay?

Yeah I thought about that. If they have an LSP configured, then I would say we should fail it simply because the LSP also helps with privacy. Maybe once we get route blinding it won't matter much and we can do the "channel or not?" check?